payloadgenerator/securerandom.js

```diff
@@ -0,0 +1,63 @@
+// Auxiliary variables/constants for payload generation.
+var SecureRandom = Java.type("java.security.SecureRandom");
+var random = new SecureRandom();
+var NUMBER_OF_PAYLOADS = 10;
+var INITIAL_VALUE = 1;
+var count = INITIAL_VALUE;
+
+/**
+ * Returns the number of generated payloads, zero to indicate unknown number.
+ * The number is used as a hint for progress calculations.
+ * 
+ * @return {number} The number of generated payloads.
+ */
+function getNumberOfPayloads() {
+	return NUMBER_OF_PAYLOADS;
+}
+
+/**
+ * Returns true if there are still payloads to generate, false otherwise.
+ * 
+ * Called before each call to next().
+ * 
+ * @return {boolean} If there are still payloads to generate.
+ */
+function hasNext() {
+	return (count <= NUMBER_OF_PAYLOADS);
+}
+
+/**
+ * Returns the next generated payload.
+ * 
+ * This method is called while hasNext() returns true.
+ * 
+ * @return {string} The next generated payload.
+ */
+function next() {
+	count++;
+	// There are other data type options offered by SecureRandom
```

Comment:
I'd remove all the other comments, we already have the templates if more info is needed.

Reply:
You mean like the function description comments or the other lines here or both?

Reply:
The function descriptions (which are in the template).